### DISCUSSION TITLE
Changing the frame of a submenu in a menu::SysMenuBar?

### DISCUSSION THREAD

**Discussion by 4JX** [fltk-rs/fltk-rs#994] [General]
Aside from `menu.set_frame()` I could only find `menu.set_down_frame()` which is not what I needed.
Specifically, this is what I want to change: https://ibb.co/0CgTnMm

**Comment by MoAlyousef** [comment]
I don't think FLTK exposes the window used for these menus. You can replace the default UpBox with a BorderBox or FlatBox, and this will apply to the menu window:
```rust
app::set_frame_type2(FrameType::UpBox, FrameType::BorderBox);
```

**Reply by 4JX** [reply]
Thanks! Will try that. 

On a completely unrelated note to not make another post, do you know how to stop widgets from being resizeable? 

As in stopping them from being resized when you put your cursor on the border of one, click and drag. I know the window and some "containers"/groups have the `make_resizable()` method but is there something similar for widgets themselves?

**Reply by MoAlyousef** [reply]
Nothing for widgets. You can override resizing by placing a widget inside a group, and marking the group as not resizable:
```rust
    let mut grp = Group::new(280, 200, 80, 40, "");
    let but = Button::new(280, 200, 80, 40, "but");
    grp.end();
    grp.make_resizable(false);
```

Or you can override resizing in the handle method or the window's resize callback:
```rust
    but.handle(|b, ev| match ev {
        Event::Resize => {
            b.set_size(80, 40);
            // or but.resize(some_x, some_y, 80, 40);
            true
        },
        _ => false,
    });
```

```rust
    wind.resize_callback(move |win, _x, _y, _w, _h| {
        but.set_size(80, 40);
    });
```

**Reply by MoAlyousef** [reply]
You can also set a size_range for your window, so it resizing it will have limits:
```rust
win.size_range(lower_w, lower_h, higher_w, higher_h);
```
Passing zero for the higher ranges means no limits for increasing the window's size.
